src/main/java/com/google/api/codegen/config/LanguageGenerator.java

```diff
@@ -132,6 +135,23 @@ public String getFormattedPackageName(String packageName) {
     }
   }
 
+  private static class NodeJSLanguageFormatter extends SimpleLanguageFormatter {
+    public NodeJSLanguageFormatter() {
+      super("-", null, false);
+    }
+
+    public String getFormattedPackageName(String packageName) {
+      List<String> nameComponents = Splitter.on(DEFAULT_PACKAGE_SEPARATOR).splitToList(packageName);
+      int size = nameComponents.size();
+      if (size < 3
+          || !nameComponents.get(0).equals("google")
+          || !nameComponents.get(size - 1).startsWith("v")) {
```

Comment:
make a static function for this if logic, and call it also from GoLanguageFormatter